### PR TITLE
Add migrate-external-ids tool

### PR DESCRIPTION
**Pull request by smatting**
This pr adds `tools/db/migrate-external-ids`

This tool fills entries in the table `spar.scim_external` (introduced in #1359 and rreleased [2021-02-16](https://github.com/wireapp/wire-server/releases/tag/v2021-02-16)) from table `spar.scim_external_ids` and `brig.user`. It is meant to be used in the migration from `spar.scim_external_ids` to `spar.scim_external`.

### CODE REVIEW

**Comment by smatting** [comment]
I tested locally with ~20 external ids created by the spar integration tests.
Tested:
- migration. all external ids are inserted
- dry-run doesn't insert any rows
- brig user with missing team id is handled and reported at the end

File: tools/db/migrate-external-ids/src/Work.hs [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+    readUserTeam Env {..} uids =
+      runClient envBrig $ do
+        query select (params Quorum (Identity uids))
+      where
+        select :: PrepQuery R (Identity [UserId]) UserTeam
+        select = "SELECT id, team FROM user where id in ?"
+
+isDebug :: Debug -> Bool
+isDebug Debug = True
+isDebug NoDebug = False
```
Comment (fisx) [inline tools/db/migrate-external-ids/src/Work.hs:138]: Move to the type definition?  (No strong opinion.)

File: tools/db/migrate-external-ids/src/Work.hs [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+  runConduit $
+    zipSources
+      (CL.sourceList [(1 :: Int32) ..])
+      (readLegacyExternalIds env)
+      .| CC.mapM (resolveTeam env)
+      .| sinkMain env
+
+  failCount <- readIORef envFailCount
+  when (failCount > 0) $
+    Log.warn envLogger (Log.msg @String (show failCount <> " external ids have *NOT* been migrated.\nAn external id fails to me migrated if the mapped user doesn't exist or doesn't have a team."))
```
Comment (fisx) [inline tools/db/migrate-external-ids/src/Work.hs:98]:
```suggestion
    Log.warn envLogger (Log.msg @String (show failCount <> " external ids have *NOT* been migrated.\nAn external id fails to be migrated if the mapped user doesn't exist or doesn't have a team."))
```

Comment (fisx) [reply tools/db/migrate-external-ids/src/Work.hs:98]: `failures :: [LegacyExternalId]` containing all failed externalIds would be more helpful.  (I'm pretty sure those fit into memory, too.)

On second thought: all the information is contained in stdout/stderr, so we can just capture that.  Furthermore, this script is idempotent, so even if we forget to capture, we can just run it again.

So this looks good to me!

**Review by fisx — APPROVED** [review]

**Comment by fisx** [comment]
@smatting good to ship!  feel free to ignore everything i write in the review, it's all nit-picks!

**Pull request merged**